examples/dmd/dg_euler.py

```diff
@@ -0,0 +1,494 @@
+glob_size = vfes.GlobalTrueVSize()
+if myid == 0:
+    print("Number of unknowns: " + str(glob_size))
+
+# 8. Define the initial conditions, save the corresponding mesh and grid
+#    functions to a file. This can be opened with GLVis with the -gc option.
+#    The solution u has components {density, x-momentum, y-momentum, energy}.
+#    These are stored contiguously in the BlockVector u_block.
+
+offsets = [k*vfes.GetNDofs() for k in range(num_equation+1)]
+offsets = mfem.intArray(offsets)
+u_block = mfem.BlockVector(offsets)
+
+#  Momentum grid function on dfes for visualization.
+mom = mfem.ParGridFunction(dfes, u_block,  offsets[1])
+
+#  Initialize the state.
+u0 = InitialCondition(num_equation)
+sol = mfem.ParGridFunction(vfes, u_block.GetData())
+sol.ProjectCoefficient(u0)
+
+smyid = '{:0>6d}'.format(myid)
+pmesh.Print("vortex-mesh."+smyid, 8)
+for k in range(num_equation):
+    uk = mfem.ParGridFunction(fes, u_block.GetBlock(k).GetData())
+    sol_name = "vortex-" + str(k) + "-init."+smyid
+    uk.Save(sol_name, 8)
+
+# 9. Set up the nonlinear form corresponding to the DG discretization of the
+#    flux divergence, and assemble the corresponding mass matrix.
+Aflux = mfem.MixedBilinearForm(dfes, fes)
+Aflux.AddDomainIntegrator(DomainIntegrator(dim))
+Aflux.Assemble()
+
+A = mfem.ParNonlinearForm(vfes)
+rsolver = RiemannSolver()
+ii = FaceIntegrator(rsolver, dim)
+A.AddInteriorFaceIntegrator(ii)
+
+# 10. Define the time-dependent evolution operator describing the ODE
+#     right-hand side, and perform time-integration (looping over the time
+#     iterations, ti, with a time-step dt).
+euler = FE_Evolution(vfes, A, Aflux.SpMat())
+
+if (visualization):
+    MPI.COMM_WORLD.Barrier()
+    sout = mfem.socketstream("localhost", 19916)
+    sout.send_text("parallel " + str(num_procs) + " " + str(myid))
+    sout.precision(8)
+    sout.send_solution(pmesh, mom)
+    sout.send_text("pause")
+    sout.flush()
+    if myid == 0:
+        print("GLVis visualization paused.")
+        print(" Press space (in the GLVis window) to resume it.")
+
+# Determine the minimum element size.
+my_hmin = 0
+if (cfl > 0):
+    my_hmin = min([pmesh.GetElementSize(i, 1) for i in range(pmesh.GetNE())])
+hmin = MPI.COMM_WORLD.allreduce(my_hmin, op=MPI.MIN)
+
+# initialize timers
+fom_timer, dmd_training_timer, dmd_prediction_timer = \
+        StopWatch(), StopWatch(), StopWatch()
+
+fom_timer.Start()
+t = 0.0
+ts = []
+euler.SetTime(t)
+ode_solver.Init(euler)
+if (cfl > 0):
+    #  Find a safe dt, using a temporary vector. Calling Mult() computes the
+    #  maximum char speed at all quadrature points on all faces.
+    z = mfem.Vector(A.Width())
+    A.Mult(sol, z)
+    max_char_speed = MPI.COMM_WORLD.allreduce(dg_euler_common.max_char_speed, op=MPI.MAX) 
+    dg_euler_common.max_char_speed = max_char_speed 
+    dt = cfl * hmin / dg_euler_common.max_char_speed / (2*order+1)
+fom_timer.Stop()
```

Comment:
In the [libROM version](https://github.com/LLNL/libROM/blob/master/examples/dmd/dg_euler.cpp), the stopping of the timer is placed before the cfl statement on line 343.

Reply:
Thanks for catching this. I have placed the fom_timer before the cfl statements. The difference in FOM time is now 1 order (instead of 2).